airflow/sensors/base.py

```diff
@@ -251,7 +251,7 @@ def run_duration() -> float:
                 raise AirflowRescheduleException(reschedule_date)
             else:
                 time.sleep(self._get_next_poke_interval(started_at, run_duration, try_number))
-                try_number += 1
+            try_number += 1
```

Comment:
@weiwenlan It would be nice to see a test case asserting the expected poke interval to `time.sleep()` method for reschedule and non-reschedule modes. :)

Reply:
@utkarsharma2 Thanks for the suggestion! I will try to add some test cases for the exponential_backoff parameter in different modes.